tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/ResourceMethodReturnTypeTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.object;
+
+import org.ballerinalang.test.BCompileUtil;
+import org.ballerinalang.test.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import static org.ballerinalang.test.BAssertUtil.validateError;
+
+/**
+ * Test cases for return types in resource functions.
+ */
+public class ResourceMethodReturnTypeTest {
+    @Test()
+    public void testResourceMethodReturnType() {
+        CompileResult compileResult = BCompileUtil.
+                compile("test-src/object/resource_method_return_type_negative.bal");
+        int i = 0;
+        validateError(compileResult, i++, "return type of the resource function does not allow " +
+                "'client object' types", 30, 58);
+        validateError(compileResult, i++, "return type of the resource function does not allow " +
+                "'client object' types", 34, 71);
+        validateError(compileResult, i++, "return type of the resource function does not allow " +
+                "'function' types", 34, 71);
+        validateError(compileResult, i++, "return type of the resource function does not allow " +
```

Comment:
I believe it may be useful to the user to see the actual type in the error message as well. Just `client object` and `function` along with position may not help the user to track the location clearly.

Reply:
I will update

Reply:
Updated 53805b7787f1ab374d8914f9642d6d8edad11b52